embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImpl.kt

```diff
@@ -1,9 +1,69 @@
 package io.embrace.android.embracesdk.capture.envelope.resource
 
+import android.content.pm.PackageInfo
+import io.embrace.android.embracesdk.BuildConfig
+import io.embrace.android.embracesdk.Embrace.AppFramework
+import io.embrace.android.embracesdk.capture.metadata.AppEnvironment
+import io.embrace.android.embracesdk.capture.metadata.HostedSdkVersionInfo
+import io.embrace.android.embracesdk.capture.metadata.MetadataService
+import io.embrace.android.embracesdk.internal.BuildInfo
+import io.embrace.android.embracesdk.internal.DeviceArchitecture
 import io.embrace.android.embracesdk.internal.payload.EnvelopeResource
 
-internal class EnvelopeResourceSourceImpl : EnvelopeResourceSource {
+internal class EnvelopeResourceSourceImpl(
+    private val hosted: HostedSdkVersionInfo,
+    private val environment: AppEnvironment.Environment,
+    private val buildInfo: BuildInfo,
+    private val packageInfo: PackageInfo,
+    private val appFramework: AppFramework,
+    private val deviceArchitecture: DeviceArchitecture,
+    private val device: Device,
+    private val metadataService: MetadataService
+) : EnvelopeResourceSource {
+
     override fun getEnvelopeResource(): EnvelopeResource {
-        throw NotImplementedError("Not yet implemented")
+        return EnvelopeResource(
+            appVersion = packageInfo.versionName.toString().trim { it <= ' ' },
```

Comment:
What does this do & is it a change in behavior from the previous implementation?

Reply:
yes, it's the same behavior. https://github.com/embrace-io/embrace-android-sdk/blob/f1752cf7319d63e151309d57a7c56a4916744895/embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/injection/EssentialServiceModule.kt#L103

I don't fully understand yet why the trim is required